### PR TITLE
API-77: add root endpoint on API

### PR DESCRIPTION
**Pull request by skeleton**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

Add a root endpoint to help userw to discover all routes in the API.

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: app/config/security.yml [hunk 1]
```diff
@@ -44,6 +44,10 @@ security:
             pattern:                        ^/api/oauth/v1/token
             security:                       false
 
+        api_endpoint:
```
Comment (phaseinducer) [inline app/config/security.yml]: `api_index` ?

File: src/Pim/Bundle/ApiBundle/Controller/RootEndpointController.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    /**
+     * @param Route $route
+     * @param array $methods
+     *
+     * @return array
+     */
+    protected function getMethods(Route $route, array $methods)
+    {
+        if (!isset($methods[$route->getPath()])) {
+            $methods[$route->getPath()] = [];
```
Comment (phaseinducer) [inline src/Pim/Bundle/ApiBundle/Controller/RootEndpointController.php]: Useless check I think, you can drop the protected imho.

File: src/Pim/Bundle/ApiBundle/Controller/RootEndpointController.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+namespace Pim\Bundle\ApiBundle\Controller;
+
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Route;
+
+/**
+ * @author    Marie Bochu <marie.bochu@akeneo.com>
```
Comment (fitn) [inline src/Pim/Bundle/ApiBundle/Controller/RootEndpointController.php]: Few phpdoc here to explain why it exists :)

File: src/Pim/Bundle/ApiBundle/Controller/RootEndpointController.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+     */
+    public function getAction(Request $request)
+    {
+        $routes = $this->loader->load(
+            __DIR__ .
+            DIRECTORY_SEPARATOR . '..' .
+            DIRECTORY_SEPARATOR . 'Resources' .
+            DIRECTORY_SEPARATOR . 'config' .
+            DIRECTORY_SEPARATOR . 'routing.yml'
+        );
```
Comment (fitn) [inline src/Pim/Bundle/ApiBundle/Controller/RootEndpointController.php]: No possibility to get routes directly from the router?

File: src/Pim/Bundle/ApiBundle/Controller/RootEndpointController.php (hunk 2 above)
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/Controller/RootEndpointController.php]: nope, whithout this check array_merge will throw an error :disappointed:

**Pull request merged**